### PR TITLE
Feature/bayesdel pipeline integration

### PR DESCRIPTION
**Pull request by izcram**
Adds bayesdel scores to the pipeline output.

### CODE REVIEW

File: pipeline/data_merging/bayesdel/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,55 @@
+FROM ubuntu:20.04
+
+RUN apt-get update && apt-get install -y \
+    curl \
+    parallel
+    #&& rm -rf /var/lib/apt/lists/*
```
Comment (zfisch) [inline pipeline/data_merging/bayesdel/Dockerfile:6]: can remove?

File: pipeline/data_merging/bayesdel/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+RUN mkdir /opt/victor
+
+WORKDIR /opt/victor
+
+RUN curl -L -o VICTOR_linux.tgz https://www.dropbox.com/s/rkfu47k8o4mh5wg/VICTOR_linux.tgz && tar zxf VICTOR_linux.tgz && rm -rf VICTOR_linux.tgz
+
+# TODO split up and do it manually
```
Comment (zfisch) [inline pipeline/data_merging/bayesdel/Dockerfile:14]: is this still a TODO?

File: pipeline/data_merging/bayesdel/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+# installing htslib (includes tabix and bgzip)
+
+WORKDIR /opt
+
+ARG HTSLIB_VERSION=1.10.2
+# TODO squash it together
```
Comment (zfisch) [inline pipeline/data_merging/bayesdel/Dockerfile:21]: still a TODO? not clear to me what this is referring to, but prob best to either do it now or remove the todo

File: pipeline/data_merging/bayesdel/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    zlib1g-dev \
+    libbz2-dev \
+    liblzma-dev \
+    libcurl4-nss-dev
+
+RUN cd htslib-$HTSLIB_VERSION && make
+
+RUN cd htslib-$HTSLIB_VERSION && make install
+
+# tabix bgzip parallel
```
Comment (zfisch) [inline pipeline/data_merging/bayesdel/Dockerfile:38]: seems like there's lots of cleanup that can be done here, just removing comments if they're not relevant or explaining why it's worth leaving them there

File: pipeline/data_merging/bayesdel/convert_merged_variants_to_vcf.py [hunk 5]
```diff
@@ -0,0 +1,42 @@
+import pandas as pd
+import click
+
+# TODO: use variant_utils?
```
Comment (zfisch) [inline pipeline/data_merging/bayesdel/convert_merged_variants_to_vcf.py:4]: same as before -- seems best to either implement the TODO or just remove it

File: pipeline/data_merging/bayesdel/convert_merged_variants_to_vcf.py [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+def sp(s):
```
Comment (zfisch) [inline pipeline/data_merging/bayesdel/convert_merged_variants_to_vcf.py:5]: add quick comment to explain what this does

File: pipeline/data_merging/bayesdel/convert_merged_variants_to_vcf.py [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+    coord_col = 'Genomic_Coordinate_hg38'
+
+    df = pd.read_csv(path, sep='\t')
+
+    s = df.apply(lambda d: sp(d[coord_col]), axis=1)
+
+    df_vcf = pd.DataFrame.from_dict(dict(zip(s.index, s.values))).T
+    df_vcf.columns = ['CHROM', 'POS', 'ID', 'REF', 'ALT']
+    df_vcf['QUAL'] = df_vcf['FILTER'] = df_vcf['ID'] = '.'
+    df_vcf['INFO'] = '.' #'Dummy1=nan;Dummy2=nan'
```
Comment (zfisch) [inline pipeline/data_merging/bayesdel/convert_merged_variants_to_vcf.py:19]: cleanup

File: pipeline/data_merging/bayesdel/convert_merged_variants_to_vcf.py [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+
+    df_vcf = df_vcf[~df_vcf['ALT'].apply(lambda s: any(x not in 'ACTG' for x in s))]
+
+    with open(vcf_path, 'w') as f:
+        # TODO split string up
```
Comment (zfisch) [inline pipeline/data_merging/bayesdel/convert_merged_variants_to_vcf.py:24]: cleanup

File: pipeline/data_merging/bayesdel/convert_merged_variants_to_vcf.py [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+        #header = '##fileformat=VCFv4.0\n##source=brcaexchange\n##reference=GRCh38\n##INFO=<ID=Dummy1,Number=.,Type=String,Description="">\n##INFO=<ID=Dummy2,Number=.,Type=String,Description="">\n'
+        header = '##fileformat=VCFv4.0\n##source=brcaexchange\n##reference=GRCh38\n'
+        f.write(header)
+        f.write("#CHROM POS ID REF ALT QUAL FILTER INFO\n".replace(' ', '\t'))
+
+    df_vcf.to_csv(vcf_path, sep='\t', mode='a', index=False, header=None)
+
+# TODO use click or argparse?
```
Comment (zfisch) [inline pipeline/data_merging/bayesdel/convert_merged_variants_to_vcf.py:32]: cleanup

File: pipeline/data_merging/bayesdel/test_convert_merged_variants_to_vcf.py [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
+def test_convert_merged_to_vcf(tmp_path):
+    vcf_path = os.path.join(TEST_DATA_DIR, 'built_with_vr_ids_test_file.tsv')
+
+    output_path = tmp_path / 'file.vcf'
+    convert_merged_variants_to_vcf.convert_merged_to_vcf(vcf_path, output_path)
+
+    with open(output_path, 'r') as f:
+        lines = f.readlines()
+
+    # TOOD: more checks
```
Comment (zfisch) [inline pipeline/data_merging/bayesdel/test_convert_merged_variants_to_vcf.py:16]: can implement or remove :)

File: pipeline/luigi/bayesdel_processing.py [hunk 11]
```diff
@@ -0,0 +1,90 @@
...
+    def run(self):
+        os.chdir(data_merging_method_dir)
+
+        wdir = self.output()['wdir'].path
+        if not os.path.exists(wdir):
+            os.mkdir(wdir)
+
+        log_file = Path(self.cfg.output_dir) / 'release' / 'artifacts' / 'victor_annotation.log'
+
+        # TODO big comment with hosting issue
```
Comment (zfisch) [inline pipeline/luigi/bayesdel_processing.py:48]: ?

File: pipeline/data_merging/bayesdel/convert_merged_variants_to_vcf.py [hunk 12]
```diff
@@ -0,0 +1,45 @@
...
+    df = pd.read_csv(path, sep='\t')
+
+    vcf_cols = df.apply(lambda d: _coord_to_vcf_cols(d[coord_col]), axis=1)
+
+    df_vcf = pd.DataFrame.from_dict(dict(zip(vcf_cols.index, vcf_cols.values))).T
+    df_vcf.columns = ['CHROM', 'POS', 'ID', 'REF', 'ALT']
+    df_vcf['QUAL'] = df_vcf['FILTER'] = df_vcf['ID'] = '.'
+    df_vcf['INFO'] = '.'
+
+    # filtering out variants with unambiguous bases
```
Comment (zfisch) [inline pipeline/data_merging/bayesdel/convert_merged_variants_to_vcf.py:25]: clarification -- isn't this filtering out variants with ambiguous bases? not unambiguous bases.

**Comment by izcram** [comment]
There was still the open question on whether or not to include the `vAnnGeneAll` column. Do you have an update on this @melissacline ? If not, and there are no opinions about this, we can leave it out, and still add it later on. My intention is to wrap this up by the next dev call.

**Comment by melissacline** [comment]
Hi @izcram - let's leave it out for now, and we can add it later as needed.  Thanks!

File: pipeline/data_merging/bayesdel/Readme.md [hunk 13]
```diff
@@ -0,0 +1,13 @@
...
+## Docker Commands
+
+```
+docker build . -t brcachallenge/victor:0.1
+docker push brcachallenge/victor:0.1
+```
+
+## Data Management
+
+The VICTOR pipeline has fairly large data dependencies. In order to allow for reproducible pipeline runs the complete data is redownloaded for every brcaexchange pipeline release. Since large parts of the data are not expected to change much over time, harddisk space can be significantly reduced by hardlinking files which are identical across different releases. This can easily be done with tools like `jdupes` (https://github.com/jbruchon/jdupes).
```
[the remark this answers is not in the document]
Comment (izcram) [reply pipeline/data_merging/bayesdel/Readme.md:13]: no, that data doesn't get bundled in to the release data. It is kept in a separate directory analogous to the data required for the seqrepo. I intend to add the `jdupes` things in a separate PR

**Pull request merged**